sdk-base/src/main/java/com/mapbox/maps/module/MapTelemetry.java

```diff
@@ -1,52 +1,67 @@
-package com.mapbox.maps.module
+package com.mapbox.maps.module;
 
-import android.os.Bundle
+import android.os.Bundle;
+
+import androidx.annotation.Nullable;
+
+import com.mapbox.common.TelemetryUtils;
 
 /**
  * Definition of map telemetry
  */
-interface MapTelemetry {
+public interface MapTelemetry {
 
   /**
    * Register the app user turnstile event
    */
-  fun onAppUserTurnstileEvent()
+  void onAppUserTurnstileEvent();
 
   /**
    * Set the end-user selected state to participate or opt-out in telemetry collection.
    */
-  fun setUserTelemetryRequestState(enabled: Boolean)
+  void setUserTelemetryRequestState(boolean enabled);
 
   /**
    * Disables a started telemetry service for this session only.
    */
-  fun disableTelemetrySession()
+  void disableTelemetrySession();
 
   /**
    * Set the debug logging enabled states.
    *
+   * @deprecated
    * Note this method is deprecated, there will be no operations when it is called.
    *
    * @param debugLoggingEnabled whether to enable the debug logging for telemetry.
    */
-  @Deprecated("setDebugLoggingEnabled has been deprecated and will do no operations")
-  fun setDebugLoggingEnabled(debugLoggingEnabled: Boolean)
+  @Deprecated
+  void setDebugLoggingEnabled(boolean debugLoggingEnabled);
 
   /**
    * Set the telemetry rotation session id interval.
    *
+   * @deprecated
    * Note this method is deprecated, there will be no operations when it is called.
    *
    * @param interval the selected session interval
    * @return true if rotation session id was updated
    */
-  @Deprecated("setSessionIdRotationInterval has been deprecated and will do no operations")
-  fun setSessionIdRotationInterval(interval: Int): Boolean
+  @Deprecated
+  boolean setSessionIdRotationInterval(int interval);
 
   /**
    * Register a performance event.
    *
    * @param data performance event data
    */
-  fun onPerformanceEvent(data: Bundle?)
+  void onPerformanceEvent(@Nullable Bundle data);
+
+  /**
+   * Get the end-user selected state to participate or opt-out in telemetry collection.
+   *
+   * @return true if end-user opted-in in telemetry collection, false otherwise.
+   */
+  default boolean getTelemetryCollectionState() {
```

Comment:
perhaps align naming with `setUserTelemetryRequestState`?

Reply:
Renamed to be aligned with existing names.